# **GAS STORAGE OPTIMIZATION**

In [1]:
import sys
sys.path.append(r"C:\Users\Elie\Elie code\Function\TTF")
import warnings
warnings.filterwarnings('ignore')
# import des fonctions
from Storage_Optimization import StorageOptimization

**Function import**

In [2]:
StorageOptimization = StorageOptimization()

**Load the df with all the forward contract price (Oct 25 -> Mars 27) since 2025**

In [3]:
TTF = StorageOptimization.get_contracts_dataframe(market='TTF', contract_type='monthly')
TTF = TTF[:-1]

Chargement des données daily depuis C:\Users\Elie\Datas\TradingView\TTF curve...
37 fichiers CSV trouvés


**Stores the df in the StorageOptimization for processing**

In [4]:
StorageOptimization.load_forward_curve_data(TTF)  

=== LOADING FORWARD CURVE DATA ===


# **STEP 1: Set storage parameters first**

In [5]:
StorageOptimization.set_storage_parameters(
    capacity_gwh=175.8,
    max_injection_gwh_per_day=5.86,
    max_withdrawal_gwh_per_day=5.86,
    fixed_cost_euros=200000
)

=== 💰 STORAGE COST BREAKDOWN ===
🏠 Fixed Cost: 0.0031 €/MWh/day
⬇️  Injection Cost: 0.1 €/MWh
⬆️  Withdrawal Cost: 0.1 €/MWh
📊 Total Variable Cost: 0.2 €/MWh

=== 🧮 COST CALCULATION FORMULA ===
Fixed Cost per Day = fixed_cost_euros/ 365 days
Fixed Cost per MWh = Fixed Cost per Day / Total Capacity in MWh
175.8 GWh capacity = 175,800 MWh
Each MWh of storage capacity costs €0.0031 per day


# **STEP 2: Launch the Backtest**

In [6]:
# Appeler la fonction et récupérer tous les résultats
results = StorageOptimization.calculate_rolling_intrinsic_value_with_costs()  


=== ROLLING INTRINSIC VALUE - WITH REAL SALES ===
💰 COSTS: Injection=0.1€/MWh, Withdrawal=0.1€/MWh, Fixed=200,000€
📅 PERIODS: - Trading=2025-01-01 to 2026-04-01
             - Storage Allocation=2026-04-01 to 2027-03-31

📊 ROLLING INTRINSIC STRATEGY - WITH REAL SALES

🎯 OVERALL PERFORMANCE:
    • Initial Value: 0 €
    • Trading Period (2025-01-01 to 2026-04-01):
       - Rebalances: 73
       - Profit: 367,070 €
    • Storage Period (2026-04-01 to 2027-03-31):
       - Rebalances: 0
       - Operational Profit: 0 €
       - Fixed Cost: -200,000 €
    • Final Intrinsic Value: 274,248 €
    • TOTAL NET PROFIT: 167,070 €

📈 REBALANCING SUMMARY:
Date         Old Val    New Val    Realized PnL Net Benefit 
------------------------------------------------------------
2025-01-08           0€     7,384€           0€       7,384€
2025-01-09       7,384€         0€      12,482€       5,098€
2025-02-12           0€    43,071€       3,516€      46,587€
2025-02-19      86,142€    79,110€      14,

**A. Current Actual Performance (From Your Backtest)**

from simulation:
- Rebalancing Profit (10 months): 367,070 €
- Number of Trades: 73
- Fixed Cost (annual): 200,000 €

Simple math:
- Net P&L so far = 367,070 € - 200,000 €
                 = **167,070 €**

**C. Projection to Full Year**

If we continue at same pace for 4 more months (until Mars 26) :

- Additional profit = 367,070 € × (2/14) = 52 438,5 €
- Total annual profit = 367,070 € + 52 438,5 € = 419 508,5 €
- Annual net P&L = 419 508,5 € - €200,000 € = **219 508,5 €**